src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsresettlementpassportapi/service/PoPUserOTPServiceTest.kt

```diff
@@ -77,4 +97,31 @@ class PoPUserOTPServiceTest {
     val result = popUserOTPService.getAllOTPs()
     Assertions.assertEquals(popUserOTPList, result)
   }
+
+  @Test
+  fun `test create Pop User OTP - creates and returns PoP User OTP`() {
```

Comment:
could we add a test for what happens if the OTP provided doesn't match

Reply:
Added the OTP not match test